app/src/test/java/com/rkm/tasky/util/storage/implementation/SessionStorageImplTest.kt

```diff
@@ -20,31 +19,34 @@ import org.junit.Assert.*
 
 import org.junit.After
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
+import org.junit.rules.TemporaryFolder
 import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
-import org.robolectric.RuntimeEnvironment
 
-private const val TEST_DATASTORE_NAME: String = "test_datastore"
+private const val TEST_DATASTORE_NAME: String = "test_datastore.preferences_pb"
 
 @OptIn(ExperimentalCoroutinesApi::class)
 @RunWith(RobolectricTestRunner::class)
 class SessionStorageImplTest {
 
-    init {
-        FakeAndroidKeyStore.setup
-    }
-
-    private val context: Context = RuntimeEnvironment.getApplication()
-    private val dispatcher = StandardTestDispatcher()
-    private val dataStore: DataStore<Preferences> = PreferenceDataStoreFactory.create(
-        produceFile = {context.preferencesDataStoreFile(TEST_DATASTORE_NAME)}
-    )
-    private val encryptor = DataStoreEncryptor()
-    private val sessionStorage = SessionStorageImpl(dataStore, encryptor)
+    @get:Rule
+    val tmpFolder = TemporaryFolder.builder().assureDeletion().build()
+    private lateinit var dispatcher: CoroutineDispatcher
+    private lateinit var dataStore: DataStore<Preferences>
+    private lateinit var encryptor: DataStoreEncryptor
+    private lateinit var sessionStorage: SessionStorageImpl
 
     @Before
     fun setUp() {
+        FakeAndroidKeyStore.setup
+        dispatcher = StandardTestDispatcher()
+        dataStore = PreferenceDataStoreFactory.create(
+            produceFile = {tmpFolder.newFile(TEST_DATASTORE_NAME)}
+        )
```

Comment:
Don’t forget to clear your `dataStore` in the `@After` annotated function using: `dataStore.clear()`